### PR TITLE
Allow Int arguments for aggregation function groupBitAnd/Or/Xor

### PR DESCRIPTION
**Pull request by exmy**
Aggregation functions such as groupBitAnd/GroupBitOr/groupBitXor currently only accept `UInt*` arguments as input. Allowing `Int*` argument would expand their applicability.

```shell
create table t (a UInt32, b Int32, c Nullable(Int32)) engine=Memory();
insert into t values (44, 44, 44), ( 28, 28, 28),  (13, 13, NULL), (85, 85, 85);

select groupBitAnd(a), groupBitAnd(b), groupBitAnd(c) from t;
┌─groupBitAnd(a)─┬─groupBitAnd(b)─┬─groupBitAnd(c)─┐
│              4 │              4 │              4 │
└────────────────┴────────────────┴────────────────┘

select groupBitOr(a), groupBitOr(b), groupBitOr(c) from t;
┌─groupBitOr(a)─┬─groupBitOr(b)─┬─groupBitOr(c)─┐
│           125 │           125 │           125 │
└───────────────┴───────────────┴───────────────┘

select groupBitXor(a), groupBitXor(b), groupBitXor(c) from t;
┌─groupBitXor(a)─┬─groupBitXor(b)─┬─groupBitXor(c)─┐
│            104 │            104 │            101 │
└────────────────┴────────────────┴────────────────┘
```

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Aggregation function groupBitAnd/Or/Xor now work on signed integer data. This makes them consistent with the behavior of scalar functions bitAnd/Or/Xor.

### CODE REVIEW

File: src/AggregateFunctions/Helpers.h [hunk 1]
```diff
@@ -87,8 +87,8 @@ static IAggregateFunction * createWithNumericType(const IDataType & argument_typ
     return nullptr;
 }
 
-template <template <typename, typename> class AggregateFunctionTemplate, template <typename> class Data, typename... TArgs>
-static IAggregateFunction * createWithUnsignedIntegerType(const IDataType & argument_type, TArgs && ... args)
+template <template <typename, typename> class AggregateFunctionTemplate, template <typename> class Data, bool support_signed = false, typename... TArgs>
+static IAggregateFunction * createWithOptionSignedIntegerType(const IDataType & argument_type, TArgs && ... args)
```
Comment (rschu1ze) [inline src/AggregateFunctions/Helpers.h:91]: To retain an uniform interface across all helper functions in this header, please don't change `createWithUnsignedIntegerType()`. Instead, please add a function `createWithSignedIntegerType()` that wraps l. 103-109. And then please make a function `createWithIntegerType()` which calls the previous two functions using a check similar to l. 134.

**Review by rschu1ze — COMMENTED** [review]
Could you please add a few stateless for the extended arguments to tests/queries/0_stateless/? (see [here](https://clickhouse.com/docs/en/development/tests))

Please also double-check that the aggregation function produces for all inputs (including edge cases) the same values as the scalar `bitAnd/Or/Xor()` functions (see [here](https://clickhouse.com/docs/en/sql-reference/functions/bit-functions#bitanda-b)).

Thanks!

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit c8cccb393e33ce1b5be5301db6d172a99da54420 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49292/c8cccb393e33ce1b5be5301db6d172a99da54420/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by exmy** [comment]
@rschu1ze Thanks for advice! Please review the new change.

File: tests/queries/0_stateless/00498_bitwise_aggregate_functions.reference [hunk 2]
```diff
@@ -2,3 +2,11 @@
 1	[1,5,9,13,17]	29	1	17
 2	[2,6,10,14,18]	30	2	18
 3	[3,7,11,15,19]	31	3	19
+0       [0,-4,-8,-12,-16]       -4      0       0
```
Comment (rschu1ze) [inline tests/queries/0_stateless/00498_bitwise_aggregate_functions.reference:5]: Please regenerate these expected results, CI job "Fast Test" currently reports an error. You can test locally with `PATH=$PATH:build/programs/ tests/./clickhouse-test 00498_bitwise_`.

Comment (rschu1ze) [reply tests/queries/0_stateless/00498_bitwise_aggregate_functions.reference:5]: (Tried to fix them by myself but I don't have permissions to push into your fork of ClickHouse.)

**Review by rschu1ze — COMMENTED** [review]
Looks generally good to me.

File: docs/en/sql-reference/aggregate-functions/reference/groupbitand.md [hunk 3]
```diff
@@ -13,11 +13,11 @@ groupBitAnd(expr)
 
 **Arguments**
 
-`expr` – An expression that results in `UInt*` type.
+`expr` – An expression that results in `UInt* or Int*` type.
```
[the remark this answers is not in the document]
Comment (exmy) [reply docs/en/sql-reference/aggregate-functions/reference/groupbitand.md]: Done

File: tests/queries/0_stateless/00498_bitwise_aggregate_functions.reference (hunk 2 above)
Comment (exmy) [reply tests/queries/0_stateless/00498_bitwise_aggregate_functions.reference:5]: Thanks! Done.

**Pull request merged**